test/dotnet-new3.UnitTests/SharedHomeDirectory.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace dotnet_new3.UnitTests
+{
+    /// <summary>
+    /// This class represents shared /tmp/RANDOM-GUID/.templateengine/dotnetcli-preview/ folder
+    /// shared between multiple unit tests in same class, this is so each test
+    /// doesn't have to install everything from 0. To save some time executing tests.
+    /// </summary>
+    public class SharedHomeDirectory : IDisposable
+    {
+        private readonly ITestOutputHelper log;
+
+        public string HomeDirectory { get; } = Helpers.CreateTemporaryFolder("Home");
+        public string HomeVariable { get; } = Helpers.HomeEnvironmentVariableName;
+
+        public SharedHomeDirectory(IMessageSink messageSink)
+        {
+            this.log = new SharedTestOutputHelper(messageSink);
+            Initialize();
+        }
+
+        void Initialize()
+        {
+            var dn3Path = Environment.GetEnvironmentVariable("DN3");
```

Comment:
`DN3` variable is also used for regular dotnet new3, should we rather use different variable here?

Reply:
We need to use `DN3` to tell `dotnet new3` where to search for packages to install from.

Reply:
That's correct, sorry for confusion.